### PR TITLE
Move validation code into the argument parser

### PR DESCRIPTION
**Pull request by theotherjimmy**
It's kind of cool that the defaults get validated with the type parameter too.
Also, the choices get pushed through the type before argparse checks it against the choices list.

### CODE REVIEW

File: project_generator/commands/__init__.py [hunk 1]
```diff
@@ -11,3 +11,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import argparse
+from os.path import exists
+
+def argparse_filestring_type(string):
```
Comment (0xc0170) [inline project_generator/commands/__init__.py]: please fix the formatting in this file

Comment (theotherjimmy) [inline project_generator/commands/__init__.py]: Was it the single line ifs? Is there a style guide you prefer?

Comment (0xc0170) [inline project_generator/commands/__init__.py]: This one shall be - PEP8 . Taken from there: ``Compound statements (multiple statements on the same line) are generally discouraged.``

Just follow the one that is in this module. We might want to turn pep8 validator and fix some problems. That can be feature request

**Comment by 0xc0170** [comment]
Please provide always more information how this was tested, what is expected output (how does it changes with this patch). For instance you added choices to the generate command. 

This info shall be within the commit message as well, one line commit for this patch does not hold much value.

**Comment by 0xc0170** [comment]
LGTM, please provide better commit message, and we will merge it

**Comment by theotherjimmy** [comment]
That commit message is a bit better.

**Comment by theotherjimmy** [comment]
LGTM: CI is done, commit message is beefed up.

**Pull request merged**